examples/shop/src/pages/Catalog.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from 'react';
+import { Image, ButtonGroup, Button, Text } from '@urban-bot/core';
+import { calculateCircularIndex } from '../utils';
+import { useProducts } from '../store/products';
+import { useBucket } from '../store/bucket';
+
+export function Catalog() {
+    const [productIndex, setProductIndex] = useState(0);
+    const [imageIndex, setImageIndex] = useState(0);
+    const { products, fetchProducts } = useProducts();
+    const { addProduct, addedProducts } = useBucket();
+
+    useEffect(() => {
+        fetchProducts();
+    }, [fetchProducts]);
+
+    if (products.length === 0) {
+        return <Text>Loading...</Text>;
+    }
+
+    const activeProduct = products[productIndex];
+
+    function previousProduct() {
+        setProductIndex(calculateCircularIndex(productIndex - 1, products.length));
+        setImageIndex(0);
+    }
+
+    function nextProduct() {
+        setProductIndex(calculateCircularIndex(productIndex + 1, products.length));
+        setImageIndex(0);
+    }
+
+    function nextImage() {
```

Comment:
Why are you using "Function Declaration" ?

Reply:
Why not?

Reply:
It's better to read to divide data and actions